src/Microsoft.PowerShell.Commands.Management/commands/management/Clipboard.cs

```diff
@@ -59,102 +75,68 @@ private static string StartProcess(
             return stdout;
         }
 
-        public static string GetText()
+        public static string GetText(Hashtable clipboardActions)
         {
-            if (_clipboardSupported == false)
-            {
-                return _internalClipboard ?? string.Empty;
-            }
+            // need errors for null clipboardActions
+            // need errors for null clipboardActions["Paste"]
+            
+            Hashtable pasteAction = (Hashtable)clipboardActions["Paste"];
 
-            string tool = string.Empty;
-            string args = string.Empty;
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                string clipboardText = string.Empty;
-                ExecuteOnStaThread(() => GetTextImpl(out clipboardText));
-                return clipboardText;
-            }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            if (pasteAction.ContainsKey("UseWindowsClipboard") && ((bool)pasteAction["UseWindowsClipboard"]))
             {
-                string waylandEnv = Environment.GetEnvironmentVariable("WAYLAND_DISPLAY");
-                bool isWayland = !(string.IsNullOrEmpty(waylandEnv));
-
-                if (isWayland)
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                 {
-                    tool = "wl-paste";
+                    throw new PlatformNotSupportedException("1REPLACEME Can't run Windows clipboard methods on non-Windows platforms!");
                 }
                 else
                 {
-                    // Previous behavior: use xclip regardless of env:WAYLAND_DISPLAY
-                    tool = "xclip";
-                    args = "-selection clipboard -out";
+                    string clipboardText = string.Empty;
+                    ExecuteOnStaThread(() => GetTextImpl(out clipboardText));
+                    return clipboardText;
                 }
             }
-            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            else if (pasteAction.ContainsKey("Command"))
             {
-                tool = "pbpaste";
+                string tool = (string)pasteAction["Command"];
+                Object[] args = (Object[])pasteAction["Arguments"];
+
+                return StartProcess(tool, args);
             }
             else
             {
-                _clipboardSupported = false;
-                return string.Empty;
+                // write-warning "PSClipboardActions.Paste.Command is not set" ???
+                return _internalClipboard ?? string.Empty;
```

Comment:
Best I could do to emulate the previous behaviour for the tests, but tests should now set `$PSClipboardActions` to `$null` to get here (also forgot to update the tests again...)